eng/pipelines/templates/stages/vmr-final-join.yml

```diff
@@ -0,0 +1,79 @@
+parameters:
+# Branch of the VMR to use (to push to for internal builds)
+- name: vmrBranch
+  type: string
+  default: $(Build.SourceBranch)
+
+- name: pool_Windows
+  type: object
+  default:
+    name: $(defaultPoolName)
+    image: $(poolImage_Windows)
+    demands: ImageOverride -equals $(poolImage_Windows)
+    os: windows
+
+- name: primaryDependentJob
+  type: string
+  default: Windows_x64
+
+- name: azureDevOpsOrg
+  type: string
+  default: dnceng
+
+- name: azureDevOpsProject
+  type: string
+  default: internal
+    
+stages:
+- stage: VMR_Final_Join
+  displayName: VMR Final Join
+  dependsOn: VMR_Vertical_Build
+  variables:
+  - ${{ if eq(variables['System.TeamProject'], 'internal') }}:
+    - group: Publish-Build-Assets
+  - ${{ else }}:
+    - name: dn-bot-dnceng-build-rw-code-rw
+      value: ''
+
+  - template: ../variables/vmr-build.yml
+    parameters:
+      vmrBranch: ${{ parameters.vmrBranch }}
+
+  jobs:
+  - job: VMR_Final_Join
+    pool: ${{ parameters.pool_Windows }}
+
+    templateContext:
+      outputs:
+      - output: pipelineArtifact
+        path: $(Build.ArtifactStagingDirectory)/artifacts
+        artifact: JoinedArtifacts
+        condition: succeededOrFailed()
+
+    steps:
+    - checkout: self
+
+    - task: DownloadPipelineArtifact@2
+      inputs:
+        artifactName: 'VerticalManifests'
+        targetPath: $(Build.ArtifactStagingDirectory)/VerticalManifests
+
+    - script: $(Build.SourcesDirectory)/build.cmd 
```

Comment:
What's the reason for making this work on Windows only? I would assume that depending on machine pool wait times, we might want to run this on Unix instead. Can we make this template work cross-plat?

Reply:
good point, with the refactoring, we'll need to be able to run the preparation step in linux too

Reply:
Done